src/controller_examples/rabbitmq_controller/exec/reconciler.rs

```diff
@@ -298,21 +298,19 @@ pub fn reconcile_core(rabbitmq: &RabbitmqCluster, resp_o: Option<Response<EmptyT
                     if found_stateful_set.is_ok(){
                         let mut new_stateful_set = found_stateful_set.unwrap();
                         // rabbitmq controller doesn't support scale down, so new replicas must be greater than or equal to old replicas
-                        if new_stateful_set.spec().is_some()
-                            && new_stateful_set.spec().unwrap().replicas().is_some()
-                            && new_stateful_set.spec().unwrap().replicas().unwrap() <= rabbitmq.spec().replicas() {
-                                new_stateful_set.set_spec(stateful_set.spec().unwrap());
-                                let req_o = KubeAPIRequest::UpdateRequest(KubeUpdateRequest {
-                                    api_resource: StatefulSet::api_resource(),
-                                    name: stateful_set.metadata().name().unwrap(),
-                                    namespace: rabbitmq.namespace().unwrap(),
-                                    obj: new_stateful_set.to_dynamic_object(),
-                                });
-                                let state_prime = RabbitmqReconcileState {
-                                    reconcile_step: RabbitmqReconcileStep::AfterUpdateStatefulSet,
-                                    ..state
-                                };
-                                return (state_prime, Some(Request::KRequest(req_o)));
+                        if new_stateful_set.metadata().owner_references_contains(rabbitmq.controller_owner_ref()) {
```

Comment:
And could you document why we want to check the owner reference here (the corner case)?